File: IJA/ukol2/ija/ija2019/homework2/maps/Coordinate.java
package ija.ija2019.homework2.maps;

public class Coordinate{

	private int x;
	private int y;

	public int getX(){
		return this.x;
	}
	public int getY(){
		return thix.y;
	}
	public static Coordinate create(int x, int y){
		if( x < 0 || y < 0) return null;
		Coordinate A = new Coordinate();
		A.x = x;
		A.y = y;
		return A;
	}
	public int diffX(Coordinate c){
		return this.x - c.getX();
	}
	public int diffY(Coordinate c){
		return this.y - c.getY();
	}
	public boolean equals(Object obj){
		if (obj == this) return true;

		else if (obj == null || obj.getClass() != 	this.getClass()) return false;

		Coordinate test = (Coordinate) obj;
		return text.getX() == this.getX() && test.getY() == 	this.getY();
	}
	//public int hashCode()



}

File: ICP/MHD2/MHD2/Coordinate.cpp
#include "Coordinate.h"

Coordinate::Coordinate(double Xx, double Yy)
{
    x = Xx;
    y = Yy;
}

double Coordinate::getX()
{
    return this->x;
}
double Coordinate::getY()
{
    return this->y;
}

bool Coordinate::equals(Coordinate *c)
{
    if(c == this) return true;
    if((c->getX() >= this->getX() && c->getX()-0.0001 <= this->getX()) || (c->getX() <= this->getX() && c->getX()+0.0001 >= this->getX()))
    {
        if((c->getY() >= this->getY() && c->getY()-0.0001 <= this->getY()) || (c->getY() <= this->getY() && c->getY()+0.0001 >= this->getY()))
            return true;
    }
        return false;
}
